src/composition-engine.js

```diff
@@ -202,14 +241,22 @@ export class CompositionEngine {
         result.bind(context.bindingContext, context.overrideContext);
 
         let work = () => {
-          return Promise.resolve(context.viewSlot.removeAll(true)).then(() => {
-            context.viewSlot.add(result);
-
-            if (context.compositionTransactionNotifier) {
-              context.compositionTransactionNotifier.done();
-            }
-
-            return result;
+          let swapStrategy;
+
+          swapStrategy = context.swapOrder in swapStrategies
+                  ? swapStrategies[context.swapOrder]
+                  : swapStrategies.after;
```

Comment:
can be streamlined to ` = swapStrategies[context.swapOrder] || swapStrategies.after`.
The `in` check is implicit since if it is not `in`, we'll get `undefined` out.